.circleci/config.yml

```diff
@@ -287,7 +282,37 @@ jobs:
       - checkout
       - *setup_remote_docker
       - *install_docker_client
-      - *docker_build
+      - run:
+          name: Build Docker image
+          command: |
+            ~/bin/docker build .
+
+  docker_push_tag:
+    <<: *container_config
+    steps:
+      - checkout
+      - *setup_remote_docker
+      - *install_docker_client
+      - run:
+          name: Build and push Docker image to DockerHub
+          command: |
+            ~/bin/docker login -u $DOCKER_USER -p $DOCKER_PASS
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?} .
+            ~/bin/docker tag ${DOCKERHUB_REPO:?}:latest ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?}
+            ~/bin/docker push ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?}
```

Comment:
I understand `docker push` changes remote state. Would it be possible to print remote state portion meant to be modified by state push before *and* after each push?

`docker images ...` may be the right command though I am not sure if its output discerns between image tagged-though-not-pushed-yet from image tagged-and-pushed.

Reply:
I am not sure if you need to push both tags, something like:
`~/bin/docker push ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?} ${DOCKERHUB_REPO:?}:latest`.

I might be wrong about this.

Reply:
I don't find simple or build-in way to do that. As I see we could use the docker registry API, but it requires authentication, thus I find it out of scope of this PR.